### PR TITLE
deps: update eigenlayer-middleware

### PR DESCRIPTION
**Pull request by neutiyoo**
This PR uses eigenlayer-middleware [`v0.2.0-holesky-rewards`](https://github.com/Layr-Labs/eigenlayer-middleware/releases/tag/v0.2.0-holesky-rewards) release which includes `RewardsCoordinator`.

### CODE REVIEW

File: contracts/script/MachServiceManagerDeployer.s.sol [hunk 1]
```diff
@@ -362,32 +370,5 @@ contract MachServiceManagerDeployer is Script {
             )
         );
         vm.stopBroadcast();
-
```
Comment (ny-alt) [inline contracts/script/MachServiceManagerDeployer.s.sol:365]: As we will store the deployment output, we don't need this output generation.

File: contracts/script/MachServiceManagerDeployer.s.sol [hunk 2]
```diff
@@ -362,32 +370,5 @@ contract MachServiceManagerDeployer is Script {
...
-        vm.serializeAddress(output, "stakeRegistryImpl", address(machServiceContract.stakeRegistryImplementation));
-        vm.serializeAddress(output, "apkRegistry", address(machServiceContract.apkRegistry));
-        vm.serializeAddress(output, "apkRegistryImpl", address(machServiceContract.apkRegistryImplementation));
-        vm.serializeAddress(output, "pauserRegistry", address(pauserRegistry));
-        vm.serializeAddress(output, "machAVSProxyAdmin", address(machAVSProxyAdmin));
-        vm.serializeAddress(output, "emptyContract", address(emptyContract));
-        vm.serializeAddress(output, "operatorStateRetriever", address(machServiceContract.operatorStateRetriever));
-        string memory finalJson = vm.serializeString(output, "object", output);
-        vm.createDir("./script/output", true);
-        vm.writeJson(finalJson, deployedMachPath);
```
Comment (neutiyoo) [inline contracts/script/MachServiceManagerDeployer.s.sol:391]: As we will store the deployment output, we don't need this output generation.

File: .gitmodules [hunk 3]
```diff
@@ -4,6 +4,3 @@
 [submodule "contracts/lib/forge-std"]
 	path = contracts/lib/forge-std
 	url = https://github.com/foundry-rs/forge-std
-[submodule "contracts/lib/eigenlayer-contracts"]
-	path = contracts/lib/eigenlayer-contracts
-	url = https://github.com/Layr-Labs/eigenlayer-contracts
```
Comment (neutiyoo) [inline .gitmodules:9]: Note that this has been removed as we can use the eigenlayer-contracts from eigenlayer-middleware dependencies

**Review by bb111189 — CHANGES_REQUESTED** [review]
This can be a quite a breaking change. 

We may still need to launch new MACH AVS (without payments) on mainnet.
Also, payment may not be release to eigenlayer mainnet anytime soon.

I think we need to develop on a different branch

**Comment by neutiyoo** [comment]
OK sure, closing this PR for now in favor of https://github.com/alt-research/mach-avs/tree/feat/rewards-coordinator

**Pull request closed without merging**